From fa4a40c7d497fa99e578495f02b5ad5e449c72b1 Mon Sep 17 00:00:00 2001
From: Yury Khrustalev <yury.khrustalev@arm.com>
Date: Mon, 16 Feb 2026 12:46:39 +0000
Subject: [PATCH 12/22] tests: aarch64: fix makefile dependencies for dlopen
 tests for BTI

Some BTI tests in the sysdeps/aarch64 directory use
test shared objects via dlopen. Due to lack of direct makefile level
dependencies on these modules these tests could be run before the
required .so files would be created. This could lead to flaky test
results when running make check with -j flag. This commit fixes it.

Reviewed-by: Wilco Dijkstra  <Wilco.Dijkstra@arm.com>
(cherry picked from commit 4db97f4d04cd2ad87f75b7219de2cb851fc62ebd)
---
 sysdeps/aarch64/Makefile | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sysdeps/aarch64/Makefile b/sysdeps/aarch64/Makefile
index 0d32eddf366cc25a9491c3de27e81ef0b12b6ba8..ff4f72c45e64c15026bb45900fc91fc2e33aa854 100644
--- a/sysdeps/aarch64/Makefile
+++ b/sysdeps/aarch64/Makefile
@@ -132,6 +132,12 @@ $(objpfx)tst-bti-permissive-transitive: $(objpfx)tst-bti-mod.so
 $(objpfx)tst-bti-ld-debug-shared: $(objpfx)tst-bti-mod.so
 $(objpfx)tst-bti-ld-debug-both: $(objpfx)tst-bti-mod-unprot.so
 
+$(objpfx)tst-bti-dlopen-imm.out: $(objpfx)tst-bti-mod-unprot.so
+$(objpfx)tst-bti-dlopen-prot.out: $(objpfx)tst-bti-mod-prot.so
+$(objpfx)tst-bti-dlopen-transitive.out: $(objpfx)tst-bti-mod.so
+$(objpfx)tst-bti-ld-debug-dlopen.out: $(objpfx)tst-bti-mod-unprot.so
+$(objpfx)tst-bti-permissive-dlopen.out: $(objpfx)tst-bti-mod-unprot.so
+
 $(objpfx)tst-bti-abort-unprot-preload.out: $(objpfx)tst-bti-mod-unprot-preload.so
 tst-bti-abort-unprot-preload-ENV = \
   GLIBC_TUNABLES=glibc.cpu.aarch64_bti=1 \
-- 
2.43.7

